lib/rules/no-action-recursion.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Prevent the code to use `this.actionCall()` from the action `actionCall`
+ */
+"use strict";
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+function reportMessage(action = "") {
+  return `Closure action '${action}' must not have same name as calling action '${action}'`;
+}
+
+module.exports = {
+  meta: {
+    docs: {
+      description:
+        "disallow closure actions to have same name as calling actions",
+      category: "Stylistic Issues",
+      recommended: false
+    },
+    reportMessage
+  },
+  create: function create(context) {
+    function get(obj, path) {
```

Comment:
I feel like this could/should be in a utils file. We have similar code within eslint-plugin-ember. Is there anything we can borrow from there?

Reply:
Fixed